Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.35568&attempt_id=af7d0902-b8e6-4d9b-8b4c-07ee251e53bc)

<details><summary>Partial log (click to expand)</summary><p>

```
        --release-urlbase= \
         --download=all --with-intl=small-icu
Usage: configure [options]

configure: error: option --dest-cpu: invalid choice: 'aarch64' (choose from 'arm', 'arm64', 'ia32', 'mips', 'mipsel', 'mips64el', 'ppc', 'ppc64', 'x32', 'x64', 'x86', 's390', 's390x')
make: *** [Makefile:664: node-v6.13.0.tar-headers] Error 2
builder for '/nix/store/mdr99p34wa40k0r8lphmq2mm8gc4j5zk-nodejs-headers-6.13.0.tar.gz.drv' failed with exit code 2
cannot build derivation '/nix/store/aj15nancvm6whjfc226p0lsqs83fmwib-nodejs-install-headers.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/2d58k6p6qax0dyk2h9qpfr0j2mg61hsw-nodejs-6.13.0.drv': 1 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/2d58k6p6qax0dyk2h9qpfr0j2mg61hsw-nodejs-6.13.0.drv' failed
```
</p></details>

--- a/pkgs/development/web/nodejs/nodejs.nix
+++ b/pkgs/development/web/nodejs/nodejs.nix
@@ -26,6 +26,11 @@ let
      */
   ]) (builtins.attrNames sharedLibDeps);
 
+  copyLibHeaders =
+    map
+      (name: "${getDev sharedLibDeps.${name}}/include/*")
+      (builtins.attrNames sharedLibDeps);
+
   extraConfigFlags = optionals (!enableNpm) [ "--without-npm" ];
 in
 
@@ -74,6 +79,9 @@ in
         mkdir -p $out/share/bash-completion/completions/
         $out/bin/npm completion > $out/share/bash-completion/completions/npm
       ''}
+
+      # install the missing headers for node-gyp
+      cp -r ${concatStringsSep " " copyLibHeaders} $out/include/node
     '';
 
     meta = {

